### PR TITLE
Adding ortho_4x12 & planck_mit layouts for KBD4X.

### PR DESCRIPTION
**Pull request by newtonapple**
Enabling ortho_4x12 & planck_mit layouts for KBD4X.

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: keyboards/kbdfans/kbd4x/kbd4x.h [hunk 1]
```diff
@@ -21,25 +21,38 @@
...
 ) \
 { \
 	{ k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b }, \
 	{ k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b }, \
 	{ k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b }, \
-	{ k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b } \
+	{ k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b }  \
+}
+
+#define LAYOUT_kc_ortho_4x12( \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd4x/kbd4x.h]: I believe the use of layout macros which automatically prepend `KC_` is discouraged - it makes it difficult to use keycodes like `MO()` etc.

Comment (nooges) [reply keyboards/kbdfans/kbd4x/kbd4x.h]: I like it though, so it should probably stay.

**Comment by newtonapple** [comment]
I added the macro and now the firmware is too large haha...  So, there are couple things I usually do.  On my personal keymaps, I usually add the following to the `config.h`
```
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
```
And add `EXTRAFLAGS += -flto` to the rules.mk.

Also, but the KBD4x also doesn't come w/ backlight.  So, turning that off would save a lot of space.  I can do that if you guys are ok with it.

**Comment by nooges** [comment]
Yeah, good idea to turn backlight off in rules.mk then.

File: keyboards/kbdfans/kbd4x/kbd4x.h (hunk 1 above)
Comment (noroadsleft) [reply keyboards/kbdfans/kbd4x/kbd4x.h]: > I like it though, so it should probably stay.

:(

**Comment by noroadsleft** [comment]
I'd actually prefer if Backlight stayed on.

Switching `CONSOLE_ENABLE` to no should save close to 6kB.

**Pull request merged**